lib/git_service/commit.rb

```diff
@@ -23,21 +23,63 @@ def rugged_diff(other_ref = parent_oids.first)
       other_commit.diff(rugged_commit)
     end
 
+    def formatted_author
+      "#{rugged_commit.author[:name]} <#{rugged_commit.author[:email]}>"
+    end
+
+    def formatted_author_date
+      rugged_commit.author[:time].to_time.strftime("%c %z")
+    end
+
+    def formatted_committer
+      "#{rugged_commit.committer[:name]} <#{rugged_commit.committer[:email]}>"
+    end
+
+    def formatted_committer_date
+      rugged_commit.committer[:time].to_time.strftime("%c %z")
+    end
+
+    def formatted_commit_stats
+      diff.file_status.map do |file, stats|
+        file_stats  = file
+        file_stats << " | "
+        file_stats << (stats[:additions].to_i + stats[:deletions].to_i).to_s
+        file_stats << " "
+        file_stats << "+" if stats[:additions].positive?
+        file_stats << "-" if stats[:deletions].positive?
+        file_stats
+      end
+    end
+
     def full_message
       message = "commit #{commit_oid}\n"
       message << "Merge: #{parent_oids.join(" ")}\n" if parent_oids.length > 1
-      message << "Author:     #{rugged_commit.author[:name]} <#{rugged_commit.author[:email]}>\n"
-      message << "AuthorDate: #{rugged_commit.author[:time].to_time.strftime("%c %z")}\n"
-      message << "Commit:     #{rugged_commit.author[:name]} <#{rugged_commit.author[:email]}>\n"
-      message << "CommitDate: #{rugged_commit.author[:time].to_time.strftime("%c %z")}\n"
+      message << "Author:     #{formatted_author}\n"
+      message << "AuthorDate: #{formatted_author_date}\n"
+      message << "Commit:     #{formatted_committer}\n"
+      message << "CommitDate: #{formatted_committer_date}\n"
       message << "\n"
       message << rugged_commit.message.indent(4)
       message << "\n"
-      diff.file_status.each do |file, stats|
-        message << " #{file} | #{stats[:additions].to_i + stats[:deletions].to_i} #{"+" if stats[:additions].positive?}#{"-" if stats[:deletions].positive?}\n"
-      end
-      message << " #{diff.status_summary}"
+      message << formatted_commit_stats.join("\n").indent(1)
+      message << "\n #{diff.status_summary}"
       message
     end
+
+    def details_hash
+      {
+        "sha"            => commit_oid,
+        "parent_oids"    => parent_oids,
+        "merge_commit?"  => parent_oids.length > 1,
+        "author"         => formatted_author,
+        "author_date"    => formatted_author_date,
+        "commit"         => formatted_committer,
+        "commit_date"    => formatted_committer_date,
+        "message"        => rugged_commit.message,
```

Comment:
Should rugged_commit.message be moved out to a help method as well since it's also used within full_message?

Reply:
I guess, though I guess I figured it wasn't really worth it compared to the others which had a bit of formatting logic attached.

Reply:
Decided to do this.  There is a separate commit for the addition.